zerver/lib/upload.py

```diff
@@ -356,7 +356,7 @@ def ensure_medium_avatar_image(self, user_profile: UserProfile) -> None:
         conn = S3Connection(settings.S3_KEY, settings.S3_SECRET_KEY)
         bucket = get_bucket(conn, bucket_name)
         key = bucket.get_key(file_path)
-        image_data = force_bytes(key.get_contents_as_string())
+        image_data = key.get_contents_as_string().encode('utf8')
```

Comment:
Hmm. I'd think that an image's binary data would often simply not be valid UTF-8 -- it's not meant to be text, or to be decoded as UTF-8. What type does this `get_contents_as_string` return -- is it in fact a string, or is the name wrong and it's a bytestring? If it's a string, I'm confused how the code could be working even now.

If it is a string, is there a parallel method like `get_contents_as_bytes` or `get_contents` that we could use to get bytes?

Reply:
`get_contents_as_string()` returns the type `bytes`. Also I did a quick grep on `get_content`, couldn't find any parallel method being used.

Reply:
OK, cool. If it is `bytes`, then we don't need `encode` -- and in fact it will give an error if we try it.

So that unit test will be good because it'd help us catch that immediately. :-)